components/pin/etape_liquid_level_sensor/definition.json

```diff
@@ -0,0 +1,12 @@
+{
+  "displayName": "Liquid Level Sensor",
+  "published": true,
+  "mode": "ANALOG",
+  "direction": "INPUT",
+  "defaultPeriod": 60,
+  "analogReadMode": "PIN_VALUE",
+  "selectReadMode": true,
+  "visualization": {
```

Comment:
@lorennorman  Requesting feedback - is this `visualization` type correct?

Reply:
I copied from the potentiometer which seemed to be the only raw analog sensor. Though I did a test using the potentiometer in wippersnapper and that wouldn't read my sensor. Not sure if I was missing something on my end or the adc on my test board was busted. Is there more documentation some where I haven't found on the definition.json options?

Reply:
> Though I did a test using the potentiometer in wippersnapper and that wouldn't read my sensor

This part is concerning. It didn't read the eTape Liquid Level sensor at all? Did you ensure it was connected to a **5V** power pin as specified on the product page?

Reply:
Just tried it again, I'm using this etap https://www.adafruit.com/product/2656
 On the product page it says "Since the sensor is a resistive divider / analog out, it is easy to read it using a microcontroller/Arduino ADC pin. Connect the red wire to 3 or 5V power, the black wire to ground, then read the analog divider output on white" so I connected red to 3V, black to GND and white to A0. 
Using the Arduino IDE and an analog example sketch I could get analog readings out to the serial port just fine. When I reinstall wippersnapper and configure a potentiometer as pin A0 I just get Value: 0.0
Is there something wippersnapper is expecting for the potentiometer circuit that I'm missing? I thought that would be a work around to just get the raw analog value.

Reply:
@brentru I did a bit more testing with this setup and Arduino's IOT cloud and I discovered what I though would fix my issue but it did not. I'm testing on an esp32 Feather and was trying to read analog input on pin A0, which when i used the Arduino IDE and a simple example program worked just fine. When I tried with WipperSnapper or Arduino IOT neither worked. Then I remembered that turning on the WiFi on an esp32 disables ADC1 and A0 is attached to ADC1 so i moved the sensor to A2 which is on ADC2 and that worked in Arduino IOT cloud with wifi but reconfiguring WipperSnapper on the same board to read a potentiometer on A2 still gave 0 as the only reading. I'm not sure if I should report a bug in the esp32 WipperSnapper board definition that you shouldn't be able to select analog pins on ADC1 in WipperSnapper because the WiFi needs to be on and those pins won't be available or if there's something else going on behind the scenes that's causing analog reads to not work across the baord since I couldn't get readings from pins on ADC2 either.

Reply:
> but reconfiguring WipperSnapper on the same board to read a potentiometer on A2 still gave 0 as the only reading. 

Do you have an Arduino test sketch for this? Could you post it below? 

>  I'm not sure if I should report a bug in the esp32 WipperSnapper board definition that you shouldn't be able to select analog pins on ADC1 in WipperSnapper because the WiFi needs to be on 

Absolutely, please do!

Reply:
Here's my Arduino IOT Cloud sketch that works: Just doing an analog read on pin A2 and the Arduino IOT Cloud library takes care of wifi and uploading the data to their server. The other code I was using to test was just the Arduino built in example Examples -> Analog -> AnalogInOutSerial . Without Wifi turned on A0 works just fine in that sketch.  The circuit is just the eTape linked above plugged into 3V, GND, and either A0 or A2 depending on the setup

/* 
  Sketch generated by the Arduino IoT Cloud Thing "Untitled"
 

  Arduino IoT Cloud Variables description

  The following variables are automatically generated and updated when changes are made to the Thing

  int counter;
  int waterLevel;

  Variables which are marked as READ/WRITE in the Cloud Thing will also have functions
  which are called when their values are changed from the Dashboard.
  These functions are generated with the Thing and added at the end of this sketch.
*/

#include "thingProperties.h"

void setup() {
  // Initialize serial and wait for port to open:
  Serial.begin(9600);
  // This delay gives the chance to wait for a Serial Monitor without blocking if none is found
  delay(1500); 

  // Defined in thingProperties.h
  initProperties();

  // Connect to Arduino IoT Cloud
  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
  
  /*
     The following function allows you to obtain more information
     related to the state of network and IoT Cloud connection and errors
     the higher number the more granular information you’ll get.
     The default is 0 (only errors).
     Maximum is 4
 */
  setDebugMessageLevel(2);
  ArduinoCloud.printDebugInfo();
}

void loop() {
  ArduinoCloud.update();
  // Your code here 
  waterLevel = analogRead(A2);
  counter++;
  delay(5000);
  
  
}

Reply:
@nickMakesMoves This is feeling more like a bug, could you file this (incl. the sketch above) over on https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/issues/new?assignees=&labels=&template=bug_report.md&title=? I can track it better over there, as the issue is actually with Wippersnapper on ESP32 and not the component. I'm going to do a bug sweep after the NeoPixel feature and want to include it.

Thanks!

Reply:
@brentru Looks like you squashed the base analog sensor reading bug! I'm still seeing some unexpected values for analog reads on an esp32 I expect values between 0 and 4096 from a normal analog read and i'm getting values ~ 15000. This is fine for my application as long as its consistent but I'm guessing there's still some bug somewhere scaling these values.  It seems to display values as i would expect on a pi pico W.  Either way the component seems to be working is there anything else I need to do to at least get the Etap in the Dev components tab on wippersnapper so I don't have to keep using the potentiometer.

thanks

Reply:
@brentru anything I can do to keep this moving to get this component merged?

Reply:
@nickMakesMoves  Sorry for the delay, I was focusing on display work. 

re: The ESP32 adc issue, could you please file this as a bug report including the expected values vs scaled values, https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/issues/new?assignees=&labels=&template=bug_report.md&title=

Since this doesn't have a driver attached to it, I am going to merge it and set it as in-development. More to follow...